source/versions/versioning.rst

```diff
@@ -0,0 +1,118 @@
+==========
+Versioning
```

Comment:
First, please let me thank you for all the work you've done so far. I do not want to detract from it and this thought process would not exist if you hadn't stepped up, so thank you!

However, this particular page doesn't feel like it's particularly useful in this form. We talk about sem-ver, what it means, then the version string, but because there is so much on here, when you look at the API version string, for example, people may have lost the link to sem ver. It is better to explain what each bit means inline for us.

I think what I say needs discussion (@SpongePowered/docs) and this is just my own opinion, but I would:

* Remove the semver section, with the exception of a tip at the top saying "Sponge follows a format of the SemVer specification in its projects. You can read about general SemVer usage at http://semver.org". There really is no point in us rehashing it.
* Say that SpongeAPI and SpongeForge/SpongeVanilla follow two different versioning policies
* Highlight the important parts of the version string as the semver part and, for impl, the minecraft version
* Explain what that means for the developer (for the API version string) and the server owner (for the impl version string) - most of this page doesn't really decribe that and that's what the versioning is really all about for us.

For the API, it needs to boil down to this:

* A change in the major number indicates that some APIs that were guaranteed in the previous version have been broken. 5->6 broken world pregen APIs, for example. 6->7 was a more significant break. A bump in the major number is always accompanied by resettign the minor number to zero.
  * Developers know that their plugins _might_ break when this changes, i.e. API 6.x plugins might not run on API 7.x.
* A change in the minor number (but not the major number) means that API calls have been added, but plugins for previous minor revisions of the same major revision will still work - _but this isn't necessarily the case the other way around_.
  * Developers know their plugins will still function on 7.x+n (target 7.x, n≥0, so 7.0 plugins work on 7.1), but not necessarily the other way around.
* `-SNAPSHOT` means that the versioned API is still in progress
  * If the minor version is 0 (e.g. 8.0-SNAPSHOT), **anything** may break before it is released and plugins cannot expect stability.
  * If the minor version is greater than zero (e.g. 7.2-SNAPSHOT), anything up to the previous minor revision is still guaranteed (in this case, 7.1), but anything added since (in this case, for what will be 7.2) **may still break**, so cannot expect stability on unreleased features.

For the impl:

* A version string with an RC identifier means "recommended build", builds we consider as a reasonable quality that we would want to run ourselves.
* We tend to bump the build to indicate bug fixes and performance improvements, or any significant changes in the config or other systems that aren't related to the API (in this sense, the "patch" part of the version string isn't really a "patch" part and so it kind of moves away from the sem-ver ideology).
* For the 7.x.y part, 7.x is the version of the API that we guarantee for plugins, y is the recommended build version for this API series.
* Plugins written for 7.0 to 7.x will work. 7.x+n (n≥1) (including any SNAPSHOT features) _may not_ work.
* Any server software that has the version `n.0.0-SNAPSHOT` where the minor and recommended build numbers are 0 (or would the recommended build release be 1, I'd have to check) and the SNAPSHOT monkier is applied, versions are bleeding builds for testing the next version of the API only and may break at any time as we refine everything about it.

Reply:
Thank you for the comment and feedback! This in-depth explanation provides an outline I can follow to get it right this time. I figured a re-write was needed; I just wasn't sure yet how to proceed. Your feedback will save me much time. Thanks again!

Reply:
To be honest, what you had wasn't bad, it just seemed to kind of meander a bit much. You were doing a good job of being descriptive, just in the wrong areas! Regardless, I know that large comments that suggest a rewrite can be hard to take, so thank you for taking it in the good faith you have! :)